packages/vite/src/node/server/index.ts

```diff
@@ -1151,7 +1152,7 @@ async function restartServer(server: ViteDevServer) {
 
     // Keep the same connect instance so app.use(vite.middlewares) works
     // after a restart in middlewareMode (.route is always '/')
-    middlewares.stack = newServer.middlewares.stack
+    middlewares.wares = newServer.middlewares.wares
```

Comment:
This is likely a big breaking change. There's a fair share of metaframeworks that uses `.stack` to manually turn off some of Vite's middlewares.

Reply:
not sure you can really avoid that without patching our instance of polka to have a "fake" `stack` (which is just an alias of `wares`)

also sounds like vite should be exposing the stack of middleware itself in future. this feels like an internal property people have ended up depending on, when really vite itself should probably have a top-level `server.stack` or `server.middlewareStack` etc

Reply:
Vite isn't depending on express API, it's from connect and we document exposing a connect instance via `viteServer.middlewares`. The main goal from that is that any connect-compatible servers, like express, can easily do something like `server.use(viteServer.middlewares)`. Whatever else exported from `viteServer.middlewares` isn't locking us to a specific server-framework IMO, except connect.